src/sage/matrix/matrix_polynomial_dense.pyx

```diff
@@ -4001,11 +4008,11 @@ cdef class Matrix_polynomial_dense(Matrix_generic_dense):
                 return matrix(self.base_ring(), 0, m)
 
             if n == 0: # early exit: kernel is identity
-                return matrix.identity(self.base_ring(), m, m)
+                return matrix.identity(self.base_ring(), m)
```

Comment:
```suggestion
                return matrix.identity(self.base_ring(),m)
```

Reply:
(note: ignored: this was just for SageDays teaching purposes)